File: app/src/main/java/com/sleticalboy/dailywork/base/BaseActivity.java
package com.sleticalboy.dailywork.base;

import android.os.Bundle;
import android.os.PersistableBundle;
import android.support.annotation.Nullable;

/**
 * Created on 18-2-1.
 *
 * @author sleticalboy
 * @version 1.0
 * @description
 */
public abstract class BaseActivity extends PermissionCheckActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        prepareWork();
        setContentView(attachLayoutId());
        initView();
        initData();
    }

    protected abstract void initData();

    protected abstract void initView();

    protected abstract int attachLayoutId();

    protected void prepareWork() {
    }

    @Override
    protected void onPause() {
        super.onPause();
    }

    @Override
    protected void onResume() {
        super.onResume();
    }

    @Override
    protected void onStart() {
        super.onStart();
    }

    @Override
    protected void onStop() {
        super.onStop();
    }

    @Override
    public void onSaveInstanceState(Bundle outState, PersistableBundle outPersistentState) {
        super.onSaveInstanceState(outState, outPersistentState);
    }

    @Override
    protected void onRestoreInstanceState(Bundle savedInstanceState) {
        super.onRestoreInstanceState(savedInstanceState);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }
}

File: app/src/main/java/com/sleticalboy/dailywork/base/BaseActivity.kt
package com.sleticalboy.dailywork.base

import android.os.Bundle
import android.os.PersistableBundle
import android.view.View

/**
 * Created on 18-2-1.
 *
 * @author sleticalboy
 * @version 1.0
 * @description
 */
abstract class BaseActivity : PermissionCheckActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        prepareWork()
        setContentView(attachLayout())
        initView()
        initData()
    }

    protected abstract fun initData()

    protected abstract fun initView()

    protected abstract fun attachLayout(): Int

    protected fun prepareWork() {}

}
